### PR TITLE
Fixed issue that allows recipe2plan to force type variable resolution

### PR DESCRIPTION
**Pull request by alxrsngrtn**

### CODE REVIEW

File: src/tools/tests/storage-key-recipe-resolver-test.ts [hunk 1]
```diff
@@ -312,6 +312,27 @@ describe('recipe2plan', () => {
         assert.isTrue(it.isResolved());
       }
     });
+    it('forces resolution of type variables', async () => {
+      const manifest = await Manifest.parse(`\
+    schema Thing 
+      name: Text
+     
+    particle Writer
+       data: writes [Thing {name}] // Could also be a singleton
```
Comment (piotrswigon) [inline src/tools/tests/storage-key-recipe-resolver-test.ts:321]: you can inline the schema and remove the explicit schema declaration above:
`data: writes [Thing {name: Text}]`

File: src/tools/tests/storage-key-recipe-resolver-test.ts [hunk 2]
```diff
@@ -312,6 +312,27 @@ describe('recipe2plan', () => {
...
+      launch startup
+      arcId writeArcId
+    recipe WritingRecipe
+      thing: create persistent 'my-handle-id' 
+      Writer
+        data: thing`);
+
+      const resolver = new StorageKeyRecipeResolver(manifest);
+      for (const it of (await resolver.resolve())) {
+        assert.isTrue(it.isResolved());
```
Comment (mmandlis) [inline src/tools/tests/storage-key-recipe-resolver-test.ts:333]: could you also verify that the type in the recipe can be resolved?

Comment (piotrswigon) [reply src/tools/tests/storage-key-recipe-resolver-test.ts:333]: doesn't recipe.isResolved() check that?

Comment (mmandlis) [reply src/tools/tests/storage-key-recipe-resolver-test.ts:330]: actually should even assert that `type.getEntitySchema()` isn't null, which is the error that Satakshi was experiencing (but the recipe was resolved).

**Pull request merged**